server/etcdserver/api/v3compactor/periodic_test.go

```diff
@@ -43,7 +44,7 @@ func TestPeriodicHourly(t *testing.T) {
 	initialIntervals, intervalsPerPeriod := tb.getRetentions(), 10
 
 	// compaction doesn't happen til 2 hours elapse
-	for i := 0; i < initialIntervals; i++ {
+	for i := 0; i < initialIntervals-1; i++ {
 		rg.Wait(1)
```

Comment:
I think we also need to check the length of the returned value of `rg.Wait(1)`. It's possible that it may return a slice with 2 elements. The number of loop should be equal to the sum of the number of returned elements.

The logic should be something like below,
```
cnt := 0
for cnt <= initialIntervals-1 {
     cnt += rg.Wait(1)
     ....
}

```

This comment applies to other places as well.

Reply:
> Check every rg.Wait(1) to ensure it returns exact 1 action.

It seems `rg.Wait(1)` returning 2 isn't an issue. My previous suggestion `cnt += rg.Wait(1)` makes more sense?